include/deal.II/lac/trilinos_vector_base.h

```diff
@@ -16,1810 +16,21 @@
-    Assert (!has_ghost_elements(), ExcGhostsPresent());
-    Assert (size() == v.size(),
-            ExcDimensionMismatch (size(), v.size()));
-    AssertIsFinite(s);
-    AssertIsFinite(a);
-
-    // We assume that the vectors have the same Map
-    // if the local size is the same and if the vectors are not ghosted
-    if (local_size() == v.local_size() && !v.has_ghost_elements())
-      {
-        Assert (this->vector->Map().SameAs(v.vector->Map())==true,
-                ExcDifferentParallelPartitioning());
-        const int ierr = vector->Update(a, *(v.vector), s);
-        AssertThrow (ierr == 0, ExcTrilinosError(ierr));
-      }
-    else
-      {
-        (*this) *= s;
-        VectorBase tmp = v;
-        tmp *= a;
-        this->add(tmp, true);
-      }
-  }
-
-
-
-  inline
-  void
-  VectorBase::scale (const VectorBase &factors)
-  {
-    // if we have ghost values, do not allow
-    // writing to this vector at all.
-    Assert (!has_ghost_elements(), ExcGhostsPresent());
-    Assert (local_size() == factors.local_size(),
-            ExcDimensionMismatch(local_size(), factors.local_size()));
-
-    const int ierr = vector->Multiply (1.0, *(factors.vector), *vector, 0.0);
-    AssertThrow (ierr == 0, ExcTrilinosError(ierr));
-  }
-
-
-
-  inline
-  void
-  VectorBase::equ (const TrilinosScalar  a,
-                   const VectorBase     &v)
-  {
-    // if we have ghost values, do not allow
-    // writing to this vector at all.
-    Assert (!has_ghost_elements(), ExcGhostsPresent());
-    AssertIsFinite(a);
-
-    // If we don't have the same map, copy.
-    if (vector->Map().SameAs(v.vector->Map())==false)
-      {
-        this->sadd(0., a, v);
-      }
-    else
-      {
-        // Otherwise, just update
-        int ierr = vector->Update(a, *v.vector, 0.0);
-        AssertThrow (ierr == 0, ExcTrilinosError(ierr));
-
-        last_action = Zero;
-      }
-
-  }
-
-
-
-  inline
-  const Epetra_MultiVector &
-  VectorBase::trilinos_vector () const
-  {
-    return static_cast<const Epetra_MultiVector &>(*vector);
-  }
-
-
-
-  inline
-  Epetra_FEVector &
-  VectorBase::trilinos_vector ()
-  {
-    return *vector;
-  }
-
-
-
-  inline
-  const Epetra_Map &
-  VectorBase::vector_partitioner () const
-  {
-    return static_cast<const Epetra_Map &>(vector->Map());
-  }
-
-
-
-  inline
-  const MPI_Comm &
-  VectorBase::get_mpi_communicator () const
-  {
-    static MPI_Comm comm;
-
-#ifdef DEAL_II_WITH_MPI
-
-    const Epetra_MpiComm *mpi_comm
-      = dynamic_cast<const Epetra_MpiComm *>(&vector->Map().Comm());
-    comm = mpi_comm->Comm();
-
-#else
-
-    comm = MPI_COMM_SELF;
-
-#endif
-
-    return comm;
-  }
-
-
-
-#endif // DOXYGEN
-
+  // All functionality from the former base class VectorBase was moved to
+  // MPI::Vector as this ended up to be the only derived class.
+  // Hence, using the class VectorBase is deprecated and we only provide
+  // an alias for backward compatibility.
+  using VectorBase = MPI::Vector;
```

Comment:
We need to change three things here:
1. We should switch to a `typedef` so that we can use `DEAL_II_DEPRECATED` at the end
2. We should include `config.h` in this file
3. We should use the standard `#warning "....."` to show that this header is deprecated

Also, this header is still included in `trilinos_precondition.h`.